assets/javascripts/discourse/widgets/discourse-reactions-actions.js.es6

```diff
@@ -249,14 +249,42 @@ export default createWidget("discourse-reactions-actions", {
               this.attrs.post.current_user_reaction &&
               this.attrs.post.current_user_reaction.id == params.reaction
             ) {
-              dropReaction(postContainer, params.reaction, () => {
-                return CustomReaction.toggle(
-                  params.postId,
-                  params.reaction
-                ).then(resolve);
-              });
+              this.collapsePanels();
+              this.dropUserReaction();
+              this.attrs.post.reaction_users_count -= 1;
+              this.attrs.post.current_user_used_main_reaction = false;
+              this.setCurrentUserReaction();
+              setTimeout(() => {
```

Comment:
should user ember.run.later

Reply:
also you should make sure the timeout won't happen after the page has changed for example.